From 053de9cd47c4b1a9516e05768e2341e9bd01fa79 Mon Sep 17 00:00:00 2001
From: Corey Berla <corey@berla.me>
Date: Sat, 10 Aug 2024 11:28:59 -0700
Subject: [PATCH] file-chooser: Don't validate internal schemes

file-chooser crashes when searching because of calls to directory_ready from
filename validator.  We shouldn't be validating anyways, it's a waste
especially in search and other internal locations where we can't save files
anyways.
---
 src/nautilus-file-chooser.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/nautilus-file-chooser.c b/src/nautilus-file-chooser.c
index b34b116013..df6fa6a48a 100644
--- a/src/nautilus-file-chooser.c
+++ b/src/nautilus-file-chooser.c
@@ -432,6 +432,12 @@ on_location_changed (NautilusFileChooser *self)
 {
     g_autoptr (NautilusDirectory) directory = NULL;
     GFile *location = nautilus_window_slot_get_location (self->slot);
+    g_autofree char *scheme = g_file_get_uri_scheme (location);
+
+    if (nautilus_scheme_is_internal (scheme))
+    {
+        return;
+    }
 
     directory = nautilus_directory_get (location);
     nautilus_filename_validator_set_containing_directory (self->validator, directory);
-- 
GitLab

